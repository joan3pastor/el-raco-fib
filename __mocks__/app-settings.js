export const default_settings = {
    "AVISOSnotif": true,
    "HORARIminimalist": false,
    "HORARIpalette": 1,
    "AVISOSsize": 14,
    "HORARIsize": 13
} 

export const minimalist_settings = {
    "AVISOSnotif": true,
    "HORARIminimalist": true,
    "HORARIpalette": 1,
    "AVISOSsize": 14,
    "HORARIsize": 13
} 

export const small_fontsize_settings = {
    "AVISOSnotif": true,
    "HORARIminimalist": true,
    "HORARIpalette": 1,
    "AVISOSsize": 12,
    "HORARIsize": 12
} 

export const changed_settings = {
    "AVISOSnotif": false,
    "HORARIminimalist": true,
    "HORARIpalette": 1,
    "AVISOSsize": 17,
    "HORARIsize": 16
} 

export const missing_settings = {
    "AVISOSnotif": false,
    "HORARIpalette": 1,
    "AVISOSsize": 17,
    "HORARIsize": 16
} 

export default settings_notif_disabled = {
    "AVISOSnotif": false,
    "HORARIminimalist": false,
    "HORARIpalette": 1,
    "AVISOSsize": 14,
    "HORARIsize": 13
}

export const empty_settings = {} 


export const paleta = ['#bae1ff','#baffc9','#ffb3ba','#ffffba','#ffdfba', '#d291bc', '#e0bbe4', '#fec8d8'];
