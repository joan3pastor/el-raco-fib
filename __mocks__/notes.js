export const notes_rob_empty = {
    ROB: [],
};

export const notes_rob = {
    ROB: [
        {
          titol: "Examen Parcial 1",
          nota: 5.6,
          weight: 0.30
        },
        {
          titol: "Examen Parcial 2",
          nota: 8,
          weight: 0.40
        },
        {
          titol: "Lab",
          nota: null,
          weight: 0.30
        },
       ],
};