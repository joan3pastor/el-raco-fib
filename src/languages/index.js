import {ca} from "./ca.js";
import {es} from "./es.js";
import {en} from "./en.js";

export default {
    ca: ca,
    es: es,
    en: en,
};