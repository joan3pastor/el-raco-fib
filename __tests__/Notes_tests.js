import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import configureStore from "redux-mock-store";
import React from "react";
import axios from "axios";

import sinon, { expectation } from "sinon"; // Use this to add spies to methods
import AxiosMockAdapter from "axios-mock-adapter"; // Use this if you need to mock REST requests

// COMPONENT TO TEST
import { Notes } from "../src/components/navigation/Notes";

// MOCKS
import {
  assignatures_buit,
  assignatures_1
} from "../__mocks__/fib-api/assignatures";
import { notes_rob, notes_rob_empty } from "../__mocks__/notes";

// LANG
import { ca } from "../src/languages/ca";
const lang = ca.notes;

// Enzyme configuration
Enzyme.configure({ adapter: new Adapter() });
const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = {}; // If a store is needed replace it here

beforeEach(() => {
  /* ... */
}); // Executed before each test

describe("Testing calculateAVG method", () => {
  // Create component wrapper
  const wrapper = shallow(
    <Notes store={mockStore(initialState)} lang={lang} />, // Add props if your component requires them
    { disableLifecycleMethods: false }
  ); // If true, render(), componetDidMount() & componetWillUnmount() will not be called

  // Extract method to test from wrapper
  let calculateAVG = wrapper.instance().calculateAVG;

  it("returns avarage of a subject with no entries.", () => {
    // Context provided to method
    context = {
      props: {
        notes: notes_rob_empty
      },
      state: {},
      setState: obj => {}
    };

    // Assert test case
    expect(calculateAVG.bind(context)("ROB")).toMatchInlineSnapshot(`
                                                Array [
                                                  0,
                                                  "0.00",
                                                  "#ff0000",
                                                ]
                                `);
  });

  it("returns avarage of a subject with no entries.", () => {
    // Context provided to method
    context = {
      props: {
        notes: notes_rob
      },
      state: {},
      setState: obj => {}
    };

    // Assert test case
    expect(calculateAVG.bind(context)("ROB")).toMatchInlineSnapshot(`
                                                Array [
                                                  69.71,
                                                  "6.97",
                                                  "#0bdb00",
                                                ]
                                `);
  });
});

describe("Testing calculateFinal method", () => {
  // Create component wrapper
  const wrapper = shallow(
    <Notes store={mockStore(initialState)} lang={lang} />, // Add props if your component requires them
    { disableLifecycleMethods: false }
  ); // If true, render(), componetDidMount() & componetWillUnmount() will not be called

  // Extract method to test from wrapper
  let calculateFinal = wrapper.instance().calculateFinal;

  it("returns final score of a subject with no entries.", () => {
    // Context provided to method
    context = {
      props: {
        notes: notes_rob_empty
      },
      state: {},
      setState: obj => {}
    };

    // Assert test case
    expect(calculateFinal.bind(context)("ROB")).toMatchInlineSnapshot(`
                                                Array [
                                                  0,
                                                  "0.00",
                                                  "#ff0000",
                                                ]
                                `);
  });

  it("returns final score of a subject with no entries.", () => {
    // Context provided to method
    context = {
      props: {
        notes: notes_rob
      },
      state: {},
      setState: obj => {}
    };

    // Assert test case
    expect(calculateFinal.bind(context)("ROB")).toMatchInlineSnapshot(`
                                                Array [
                                                  48.8,
                                                  "4.88",
                                                  "#ff0000",
                                                ]
                                `);
  });
});

describe("Testing calculateProgress method", () => {
  // Create component wrapper
  const wrapper = shallow(
    <Notes store={mockStore(initialState)} lang={lang} />, // Add props if your component requires them
    { disableLifecycleMethods: false }
  ); // If true, render(), componetDidMount() & componetWillUnmount() will not be called

  // Extract method to test from wrapper
  let calculateProgress = wrapper.instance().calculateProgress;

  it("returns the progress of a subject with no entries.", () => {
    // Context provided to method
    context = {
      props: {
        notes: notes_rob_empty
      },
      state: {},
      setState: obj => {}
    };

    // Assert test case
    expect(calculateProgress.bind(context)("ROB")).toMatchInlineSnapshot(`
                                          Array [
                                            0,
                                            "0%",
                                          ]
                            `);
  });

  it("returns the progress of a subject with no entries.", () => {
    // Context provided to method
    context = {
      props: {
        notes: notes_rob
      },
      state: {},
      setState: obj => {}
    };

    // Assert test case
    expect(calculateProgress.bind(context)("ROB")).toMatchInlineSnapshot(`
                                          Array [
                                            70,
                                            "70%",
                                          ]
                            `);
  });
});

describe("Testing verifyAndSaveForm method", () => {
  // Create component wrapper
  const wrapper = shallow(
    <Notes store={mockStore(initialState)} lang={lang} />, // Add props if your component requires them
    { disableLifecycleMethods: false }
  ); // If true, render(), componetDidMount() & componetWillUnmount() will not be called

  // Extract method to test from wrapper
  let verifyAndSaveForm = wrapper.instance().verifyAndSaveForm;

  it("returns error when entry name not inserted", () => {
    // Context provided to method

    const alert_stub = jest.fn(msg => msg);
    global.alert = alert_stub;

    context = {
      props: {
        notes: notes_rob
      },
      state: {
        addNom: null,
        addWeight: "0.2",
        addNota: "7.2",
        addEntryOpened: "ROB",
        addIndex: 0
      },
      setState: obj => {
        /* ... */
      }
    };

    // Execute method
    verifyAndSaveForm.bind(context)(0);

    // Assert test case
    expect(alert_stub.mock.calls[0][0]).toMatchInlineSnapshot(
      `"ERROR: Nom no introduït."`
    );
  });

  it("returns error when entry weight not inserted", () => {
    // Context provided to method

    const alert_stub = jest.fn(msg => msg);
    global.alert = alert_stub;

    context = {
      props: {
        notes: notes_rob
      },
      state: {
        addNom: "Examen Parcial 1",
        addWeight: null,
        addNota: "7.2",
        addEntryOpened: "ROB",
        addIndex: 0
      },
      setState: obj => {
        /* ... */
      }
    };

    // Execute method
    verifyAndSaveForm.bind(context)(0);

    // Assert test case
    expect(alert_stub.mock.calls[0][0]).toMatchInlineSnapshot(
      `"ERROR: Pes no introduït."`
    );
  });

  it("returns error when entry weight out of range (0 - 1)", () => {
    // Context provided to method

    const alert_stub = jest.fn(msg => msg);
    global.alert = alert_stub;

    context = {
      props: {
        notes: notes_rob
      },
      state: {
        addNom: "Examen Parcial 1",
        addWeight: "2",
        addNota: "7.2",
        addEntryOpened: "ROB",
        addIndex: 0
      },
      setState: obj => {
        /* ... */
      }
    };

    // Execute method
    verifyAndSaveForm.bind(context)(0);

    // Assert test case
    expect(alert_stub.mock.calls[0][0]).toMatchInlineSnapshot(
      `"ERROR: Pes no vàlid. Entre 0 i 1."`
    );
  });

  it("returns error when the subject total weight surpases 100%", () => {
    // Context provided to method

    const alert_stub = jest.fn(msg => msg);
    global.alert = alert_stub;

    context = {
      props: {
        notes: notes_rob
      },
      state: {
        addNom: "Examen Parcial 1",
        addWeight: "0.7",
        addNota: "7.2",
        addEntryOpened: "ROB",
        addIndex: 0
      },
      setState: obj => {
        /* ... */
      }
    };

    // Execute method
    verifyAndSaveForm.bind(context)(0);

    // Assert test case
    expect(alert_stub.mock.calls[0][0]).toMatchInlineSnapshot(
      `"ERROR: El pes total de l'assignatura supera el 100%."`
    );
  });

  it("returns error when entry score out of range (0 - 10)", () => {
    // Context provided to method

    const alert_stub = jest.fn(msg => msg);
    global.alert = alert_stub;

    context = {
      props: {
        notes: notes_rob
      },
      state: {
        addNom: "Examen Parcial 1",
        addWeight: "0.2",
        addNota: "12",
        addEntryOpened: "ROB",
        addIndex: 0
      },
      setState: obj => {
        /* ... */
      }
    };

    // Execute method
    verifyAndSaveForm.bind(context)(0);

    // Assert test case
    expect(alert_stub.mock.calls[0][0]).toMatchInlineSnapshot(
      `"ERROR: Nota no vàlilda. Entre 0 i 10 o nul."`
    );
  });

  it("saves new entry when data is valid", () => {
    // Context provided to method

    const notesToReduxStub = jest.fn(msg => msg);

    context = {
      props: {
        notes: notes_rob,
        notesToRedux: notesToReduxStub
      },
      state: {
        addNom: "Examen Parcial 1",
        addWeight: "0.2",
        addNota: "7.2",
        addEntryOpened: "ROB",
        addIndex: 0
      },
      setState: obj => {
        /* ... */
      }
    };

    // Execute method
    verifyAndSaveForm.bind(context)(0);

    // Assert test case
    expect(notesToReduxStub.mock.calls[0]).toMatchInlineSnapshot(`
      Array [
        Object {
          "ROB": Array [
            Object {
              "nota": 7.2,
              "titol": "Examen Parcial 1",
              "weight": 0.2,
            },
            Object {
              "nota": 8,
              "titol": "Examen Parcial 2",
              "weight": 0.4,
            },
            Object {
              "nota": null,
              "titol": "Lab",
              "weight": 0.3,
            },
          ],
        },
      ]
    `);
  });
});

describe("Testing Notes component", () => {

  it("Renders correctly when 'assigs' and 'notes' are NOT provided", () => {
    // Create component wrapper
    const wrapper = shallow(
      <Notes store={mockStore(initialState)} lang={lang} />, 
      { disableLifecycleMethods: false }
    ); 

    expect(wrapper).toMatchSnapshot();
  });

  it("Renders correctly when 'assigs' and 'notes' are provided", () => {
    // Create component wrapper
    const wrapper = shallow(
      <Notes store={mockStore(initialState)} lang={lang} assig={assignatures_1} notes={notes_rob} />, 
      { disableLifecycleMethods: false }
    ); 

    expect(wrapper).toMatchSnapshot();
  });
});