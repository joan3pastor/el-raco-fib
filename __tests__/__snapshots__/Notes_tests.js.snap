// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Notes component Renders correctly when 'assigs' and 'notes' are NOT provided 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fafafa",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
      No estàs cursant cap assignatura.  
  </Text>
</View>
`;

exports[`Testing Notes component Renders correctly when 'assigs' and 'notes' are provided 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fdfdfd",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#064283",
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 14,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <PercentageCircle
        bgColor="#064283"
        borderWidth={7}
        color="#3399FF"
        containerStyle={null}
        percent={77.33}
        radius={26}
        shadowColor="#fff"
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 13,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
            7.73  
        </Text>
      </PercentageCircle>
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 4,
            "textAlign": "center",
          }
        }
      >
          Nota mitja  
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <PercentageCircle
        bgColor="#064283"
        borderWidth={7}
        color="#3399FF"
        containerStyle={null}
        percent={46.400000000000006}
        radius={26}
        shadowColor="#fff"
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 13,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
            4.64  
        </Text>
      </PercentageCircle>
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 4,
            "textAlign": "center",
          }
        }
      >
          Nota total  
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <PercentageCircle
        bgColor="#064283"
        borderWidth={7}
        color="#3399FF"
        containerStyle={null}
        percent={60.00000000000001}
        radius={26}
        shadowColor="#fff"
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 13,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
            60%  
        </Text>
      </PercentageCircle>
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 4,
            "textAlign": "center",
          }
        }
      >
          Progrés  
      </Text>
    </View>
  </View>
  <ScrollViewMock>
    <View
      key="ROB"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#f2f2f2",
          "borderRadius": 5,
          "borderWidth": 0.5,
          "elevation": 3,
          "marginHorizontal": 30,
          "marginTop": 15,
        }
      }
    >
      <View
        style={Object {}}
      >
        <TouchableOpacity
          activeOpacity={0.9}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "backgroundColor": "#bae1ff",
                "borderBottomColor": "gray",
                "borderBottomWidth": 1,
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "width": 695,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#505050",
                    "fontWeight": "bold",
                    "padding": 10,
                  }
                }
              >
                Robótica   
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-around",
                "marginHorizontal": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "padding": 8,
                  "paddingTop": 12,
                }
              }
            >
              <PercentageCircle
                bgColor="#fff"
                borderWidth={5}
                color="#0bdb00"
                containerStyle={null}
                percent={77.33}
                radius={20}
                shadowColor="#d7d7d7"
              >
                <Text
                  style={
                    Object {
                      "fontSize": 12,
                      "textAlign": "center",
                    }
                  }
                >
                    7.73  
                </Text>
              </PercentageCircle>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                    "textAlign": "center",
                  }
                }
              >
                  Nota mitja  
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "padding": 8,
                  "paddingTop": 12,
                }
              }
            >
              <PercentageCircle
                bgColor="#fff"
                borderWidth={5}
                color="#ff0000"
                containerStyle={null}
                percent={46.400000000000006}
                radius={20}
                shadowColor="#d7d7d7"
              >
                <Text
                  style={
                    Object {
                      "fontSize": 12,
                      "textAlign": "center",
                    }
                  }
                >
                    4.64  
                </Text>
              </PercentageCircle>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                    "textAlign": "center",
                  }
                }
              >
                  Nota total  
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "padding": 8,
                  "paddingTop": 12,
                }
              }
            >
              <PercentageCircle
                bgColor="#fff"
                borderWidth={5}
                color="#3399FF"
                containerStyle={null}
                percent={60.00000000000001}
                radius={20}
                shadowColor="#d7d7d7"
              >
                <Text
                  style={
                    Object {
                      "fontSize": 12,
                      "textAlign": "center",
                    }
                  }
                >
                    60%  
                </Text>
              </PercentageCircle>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                    "textAlign": "center",
                  }
                }
              >
                  Progrés  
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        style={Object {}}
      >
        <Collapsible
          align="top"
          collapsed={true}
          collapsedHeight={0}
          duration={300}
          easing="easeOutCubic"
          enablePointerEvents={false}
          onAnimationEnd={[Function]}
        >
          <View
            key="Examen Parcial 10ROB"
            style={
              Object {
                "backgroundColor": "#f0f0f0",
                "borderRadius": 6,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginHorizontal": 10,
                "marginVertical": 5,
                "padding": 10,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Examen Parcial 1  
              </Text>
              <Text
                style={Object {}}
              >
                (20%)    Nota: 7.20  
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#505050"
                  name="pencil"
                  size={22}
                  style={
                    Object {
                      "marginTop": 7,
                    }
                  }
                />
              </TouchableOpacity>
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#505050"
                  name="trash"
                  size={22}
                  style={
                    Object {
                      "marginLeft": 18,
                      "marginRight": 5,
                      "marginTop": 7,
                    }
                  }
                />
              </TouchableOpacity>
            </View>
          </View>
          <View
            key="Examen Parcial 21ROB"
            style={
              Object {
                "backgroundColor": "#f0f0f0",
                "borderRadius": 6,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginHorizontal": 10,
                "marginVertical": 5,
                "padding": 10,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Examen Parcial 2  
              </Text>
              <Text
                style={Object {}}
              >
                (40%)    Nota: 8.00  
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#505050"
                  name="pencil"
                  size={22}
                  style={
                    Object {
                      "marginTop": 7,
                    }
                  }
                />
              </TouchableOpacity>
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#505050"
                  name="trash"
                  size={22}
                  style={
                    Object {
                      "marginLeft": 18,
                      "marginRight": 5,
                      "marginTop": 7,
                    }
                  }
                />
              </TouchableOpacity>
            </View>
          </View>
          <View
            key="Lab2ROB"
            style={
              Object {
                "backgroundColor": "#f0f0f0",
                "borderRadius": 6,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginHorizontal": 10,
                "marginVertical": 5,
                "padding": 10,
              }
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Lab  
              </Text>
              <Text
                style={Object {}}
              >
                (30%)    Nota: -  
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#505050"
                  name="pencil"
                  size={22}
                  style={
                    Object {
                      "marginTop": 7,
                    }
                  }
                />
              </TouchableOpacity>
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#505050"
                  name="trash"
                  size={22}
                  style={
                    Object {
                      "marginLeft": 18,
                      "marginRight": 5,
                      "marginTop": 7,
                    }
                  }
                />
              </TouchableOpacity>
            </View>
          </View>
          <View
            style={
              Object {
                "margin": 10,
              }
            }
          >
            <Button
              onPress={[Function]}
              title="AFEGIR ENTRADA"
            />
          </View>
        </Collapsible>
      </View>
    </View>
    <Text />
  </ScrollViewMock>
  <ReactNativeModal
    animationIn="slideInUp"
    animationInTiming={300}
    animationOut="slideOutDown"
    animationOutTiming={300}
    avoidKeyboard={false}
    backdropColor="black"
    backdropOpacity={0.7}
    backdropTransitionInTiming={300}
    backdropTransitionOutTiming={300}
    coverScreen={true}
    customBackdrop={null}
    deviceHeight={null}
    deviceWidth={null}
    hasBackdrop={true}
    hideModalContentWhileAnimating={false}
    isVisible={false}
    onBackButtonPress={[Function]}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalShow={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "margin": 0,
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    useNativeDriver={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 8,
            "height": 330,
            "padding": 14,
            "width": 670,
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#606060",
                "fontSize": 16,
                "fontWeight": "400",
              }
            }
          >
            Nom acte avaluatiu*:  
          </Text>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "#f2f2f2",
                "borderColor": "lightgray",
                "borderRadius": 3,
                "borderWidth": 1,
                "height": 40,
              }
            }
            underlineColorAndroid="transparent"
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#606060",
                "fontSize": 16,
                "fontWeight": "400",
              }
            }
          >
            Pes (0 a 1)*:  
          </Text>
          <TextInput
            allowFontScaling={true}
            keyboardType="decimal-pad"
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "#f2f2f2",
                "borderColor": "lightgray",
                "borderRadius": 3,
                "borderWidth": 1,
                "height": 40,
              }
            }
            underlineColorAndroid="transparent"
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#606060",
                "fontSize": 16,
                "fontWeight": "400",
              }
            }
          >
            Nota:  
          </Text>
          <TextInput
            allowFontScaling={true}
            keyboardType="decimal-pad"
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "backgroundColor": "#f2f2f2",
                "borderColor": "lightgray",
                "borderRadius": 3,
                "borderWidth": 1,
                "height": 40,
              }
            }
            underlineColorAndroid="transparent"
            value={null}
          />
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-around",
              "marginTop": 26,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.9}
            onPress={[Function]}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#dc3545",
                  "borderRadius": 6,
                  "elevation": 4,
                  "justifyContent": "center",
                  "padding": 6,
                  "width": 110,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#fff",
                    "fontWeight": "bold",
                  }
                }
              >
                 CANCEL·LAR 
              </Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.9}
            onPress={[Function]}
          >
            <View
              style={
                Object {
                  "backgroundColor": "#28a745",
                  "borderRadius": 6,
                  "elevation": 4,
                  "justifyContent": "center",
                  "padding": 6,
                  "width": 110,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#fff",
                    "fontWeight": "bold",
                  }
                }
              >
                 GUARDAR 
              </Text>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  </ReactNativeModal>
</View>
`;
